syntax = "proto3";

package bff;

service BffService {
  rpc Process (BffRequest) returns (BffResponse);
}

message BffRequest {
  string request_type = 1;
  bool use_deadline = 2;
  bool use_circuit_breaker = 3;
  bool use_backpressure = 4;
}

message BffResponse {
  string result = 1;
  bool success = 2;
  string error_message = 3;
}